### ISSUE TITLE
Incorrect caching of current build

### ISSUE THREAD

**Issue by tractorcow** [silverstripe-labs/silverstripe-travis-support#18]
With the below in `.travis.yml` the current build is cached internally, and force-pushes up to that branch will not result in those changes being detected by composer; Instead the cached archive is re-installed.

```
cache:
  directories:
    - $HOME/.composer/cache
```

We need to implement some kind of cache-buster when generating the archive and composer.json for each build.

**Comment by tractorcow** [comment]
LIkely https://github.com/composer/composer/blob/master/src/Composer/Downloader/FileDownloader.php#L243 is where the necessary cache key needs to be `busted`

I need to debug this locally to detect where the target build is being cached though.

**Comment by dhensby** [comment]
This is fixed

**Comment by tractorcow** [comment]
yeah but now we have the problem of composer timing out to resolve.

The real fix is to get it to cache everything EXCEPT the build target.

**Comment by dhensby** [comment]
Are we getting composer timing out? Where?
